homeassistant/components/plum_lightpad/__init__.py

```diff
@@ -26,56 +26,45 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+PLATFORMS = ["light"]
 
-async def async_setup(hass, config):
+
+async def async_setup(hass: HomeAssistant, config: dict):
     """Plum Lightpad Platform initialization."""
+    if DOMAIN not in config:
+        return True
 
     conf = config[DOMAIN]
-    plum = Plum(conf[CONF_USERNAME], conf[CONF_PASSWORD])
 
-    hass.data[DOMAIN] = plum
+    _LOGGER.info("Found Plum Lightpad configuration in config, importing...")
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=deepcopy(conf)
+        )
+    )
 
-    def cleanup(event):
-        """Clean up resources."""
-        plum.cleanup()
+    return True
 
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, cleanup)
 
-    cloud_web_sesison = async_get_clientsession(hass, verify_ssl=True)
-    await plum.loadCloudData(cloud_web_sesison)
-
-    async def new_load(device):
-        """Load light and sensor platforms when LogicalLoad is detected."""
-        await asyncio.wait(
-            [
-                hass.async_create_task(
-                    discovery.async_load_platform(
-                        hass, "light", DOMAIN, discovered=device, hass_config=conf
-                    )
-                )
-            ]
-        )
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Plum Lightpad from a config entry."""
+    _LOGGER.info("Setting up config entry with ID = %s", entry.unique_id)
```

Comment:
No need to make this `info` level
```suggestion
    _LOGGER.debug("Setting up config entry with ID = %s", entry.unique_id)
```